src/chaat/errors.clj

```diff
@@ -0,0 +1,20 @@
+(ns chaat.errors)
+
+(defn do-or-error
+  "Checks the result of a previous operation before performing the next 
+   operation. Will be used in a let block. If there is an error, the error will 
+   cascade down the remaining expressions in the let block without performing 
+   the remaining operations.
+
+   In situations where function f does not return a result map, it is usually 
+   when the operation is not a validation or db query. Here, one may not be 
+   concerned about checking that particular output of f before continuing, and 
+   we use the result map of the previous validation/query to continue the chain."
+  [{:keys [result error]} f & args]
+  (if (nil? error)
+    (apply f args)
+    {:result result :error error}))
+
+;; consider creating standard error messages. maybe in a map structure.
+;; (def invalid-username "Wrong username format")
+;; (def invalid-password "Wrong password format")
```

Comment:
[nit] Use bug tracking/project management tools over code comments for things like these. It's much easier to keep track of these chores that way.

Reply:
Agreed, I will slowly explore tools to help with this.
I do maintain a document of sorts for myself with chores, but I put things that I wanted to discuss during pairing sessions in the code itself. I should have either resolved or cleaned this up before the PR.

Reply:
GitHub issues should be good enough for now. I noticed you are already tracking stories on it.

Reply:
But I'll defer this to whatever your onboarders decide.